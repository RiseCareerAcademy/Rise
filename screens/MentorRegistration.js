import React from "react";
import {
  Text,
  Container,
  Content,
  Form,
  Item,
  Label,
  Input
} from "native-base";
import {
  StyleSheet,
  TouchableOpacity,
  View,
  Image,
  Button,
  ActivityIndicator
} from "react-native";
import { connect } from "react-redux";

import { registerMentor } from "../actions/user.actions";
import { DOMAIN } from "../config/url";

export class MentorRegistration extends React.Component {
  state = {
    email: "",
    password: "",
    confirmedPassword: "",
    errors: [],
    skills: "",
    profession: "",
    name: "",
    city: "",
    state: "",
    image: null,
    lastName: "",
    zipcode: ""
  };

  handlePassword = text => {
    this.setState({ password: text });
  };
  handleConfirmedPassword = text => {
    this.setState({ confirmedPassword: text });
  };
  handleSkills = text => {
    this.setState({ skills: text });
  };
  handleProfession = text => {
    this.setState({ profession: text });
  };
  handleZipCode = text => {
    this.setState({ zipcode: text });
  };

  handleEmail = text => {
    this.setState({ email: text });
  };

  handleOccupation = text => {
    this.setState({ profession: text });
  };

  handleFirstName = text => {
    this.setState({ first_name: text });
  };

  handleLastName = text => {
    this.setState({ last_name: text });
  };

  handleBiography = text => {
    this.setState({ biography: text });
  };

  validate = (password, confirmedPassword, skills, zipcode) => {
    // we are going to store errors for all fields
    // in a signle array
    const errors = [];
    if (
      skills.length == 0 ||
      profession.length == 0 ||
      name.length == 0 ||
      email.length == 0 ||
      password.length == 0 ||
      confirmedPassword.length == 0 ||
      lastName.length == 0 ||
      zipcode.length == 0
    ) {
      errors.push("All fields must be filled");
    } else if (email.length < 5) {
      errors.push("Email should be at least 5 charcters long");
    } else if (email.split("").filter(x => x === "@").length !== 1) {
      errors.push("Email should contain one @");
    } else if (email.indexOf(".") === -1) {
      errors.push("Email should contain at least one dot");
    } else if (password.length < 6) {
      errors.push("Password should be at least 6 characters long");
    } else if (password != confirmedPassword) {
      errors.push(
        "Password doesn't match" + password + " " + confirmedPassword
      );
    } else if (zipcode.length != 5 && /^\d+$/.test(zipcode)) {
      errors.push("zipcode must contain only numbers and be 5 characters long");
    }
    if (errors.length == 0) {
      // alert(errors);
      return true;
    } else {
      return false;
    }
  };

  handleSubmit = () => {
    const valid = this.validate(
      this.state.password,
      this.state.confirmedPassword,
      this.state.skills,
      this.state.zipcode
    );
    if (!valid && process.env.NODE_ENV !== "development") {
      return;
    }

    const mentor = {
      first_name: this.state.first_name || this.props.first_name,
      last_name: this.state.last_name || this.props.last_name,
      email_address: this.state.email_address || this.props.email_address,
      biography: this.state.biography || this.props.biography,
      zipcode: this.state.zipcode,
      date_of_birth: "12/24/1996",
      skills: this.state.skills,
      hobbies: "fake hobbies",
      occupation: this.state.occupation || this.props.occupation,
      password: this.state.password,
      image: this.props.profile_pic_URL
    };

    this.props.registerMentor(mentor);
  };

  render() {
    const { navigate } = this.props.navigation;
    return (
      <Container style={styles.container}>
        <Content>
          <Text style={styles.error}>
            {!this.props.registering && this.props.error}
          </Text>
          <Form>
            <Image
              style={styles.userImage}
              source={{
                uri: this.props.profile_pic_URL
              }}
            />

            <Item stackedLabel>
              <Label>Email</Label>
              <Input placeholder={this.props.email_address} onChange={this.handleEmail} />
            </Item>
            <Item stackedLabel>
              <Label>Password</Label>
              <Input
                onChangeText={this.handlePassword}
                secureTextEntry={true}
              />
            </Item>
            <Item stackedLabel>
              <Label>Confirm Password</Label>
              <Input
                onChangeText={this.handleConfirmedPassword}
                secureTextEntry={true}
              />
            </Item>
            <Item stackedLabel>
              <Label>Skills</Label>
              <Input
                onChangeText={this.handleSkills}
              />
            </Item>
            <Item stackedLabel last>
              <Label>Profession</Label>
              <Input placeholder={this.props.occupation} onChange={this.handleOccupation} />
            </Item>
            <Item stackedLabel last>
              <Label>Name</Label>
              <Input placeholder={this.props.first_name} onChange={this.handleFirstName} />
            </Item>
            <Item stackedLabel last>
              <Label>lastName</Label>
              <Input placeholder={this.props.last_name} onChange={this.handleLastName} />
            </Item>
            <Item stackedLabel last>
              <Label>zipcode</Label>
              <Input
                onChangeText={this.handleZipCode}
              />
            </Item>
            <Item stackedLabel last>
              <Label>Biography</Label>
              <Input placeholder={this.props.biography} onChange={this.handleBiography} />
            </Item>
          </Form>
        </Content>
        {this.props.registering ? (
          <ActivityIndicator animating size="large" />
        ) : (
          <TouchableOpacity
            style={styles.submitButton}
            onPress={this.handleSubmit}
          >
            <Text style={styles.submitButtonText}> Next </Text>
          </TouchableOpacity>
        )}
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 23
  },
  userImage: {
    borderRadius: 85,
    borderWidth: 3,
    height: 170,
    marginBottom: 15,
    width: 170
  },
  input: {
    margin: 15,
    height: 40,
    borderColor: "#000000",
    borderWidth: 1
  },
  submitButton: {
    backgroundColor: "#000000",
    padding: 10,
    margin: 15,
    height: 40
  },
  submitButtonText: {
    color: "white"
  },
  buttonContainer: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  center: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  },
  buttonStyle: {
    margin: 10
  },
  greyText: {
    color: "grey"
  },
  contentContainer: {
    paddingTop: 30
  },
  container: {
    margin: 5
  },
  uploadBtnContainer: {
    margin: "auto"
  },
  uploadBtn: {
    margin: "auto"
  },
  error: {
    color: "red"
  }
});

const mapStateToProps = state => ({
  loggedIn: state.user.loggedIn,
  registering: state.user.registering,
  error: state.user.error,
  email_address: state.user.email_address,
  first_name: state.user.first_name,
  occupation: state.user.occupation,
  last_name: state.user.last_name,
  biography: state.user.biography,
  profile_pic_URL: state.user.profile_pic_URL,
  error: state.user.error,
});

export default connect(
  mapStateToProps,
  { registerMentor }
)(MentorRegistration);
