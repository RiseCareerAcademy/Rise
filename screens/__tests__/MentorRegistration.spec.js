import "react-native";
import React from "react";
import { MentorRegistration } from "../MentorRegistration";
import { shallow } from "enzyme";
global.alert = jest.fn();
const props = {
  navigation: {
    navigate: () => {},
  },
};

const setupInstance = () =>
  shallow(<MentorRegistration {...props} />).instance();

it("returns true if email state is changed", () => {
  const instance = setupInstance();
  instance.handleEmail("user@email.com");
  expect(instance.state.email_address).toEqual("user@email.com");
});

it("returns true if password state is changed", () => {
  const instance = setupInstance();
  instance.handlePassword("password");
  expect(instance.state.password).toEqual("password");
});

it("returns true if confirmPassword state is changed", () => {
  const instance = setupInstance();
  instance.handleConfirmedPassword("password");
  expect(instance.state.confirmedPassword).toEqual("password");
});

it("returns true if skill state is changed", () => {
  const instance = setupInstance();
  instance.handleSkills("skill");
  expect(instance.state.skills).toEqual("skill");
});

it("returns true if profession state is changed", () => {
  const instance = setupInstance();
  instance.handleProfession("profession");
  expect(instance.state.profession).toEqual("profession");
});

it("returns true if first name state is changed", () => {
  const instance = setupInstance();
  instance.handleFirstName("name");
  expect(instance.state.first_name).toEqual("name");
});

it("returns true if last name state is changed", () => {
  const instance = setupInstance();
  instance.handleLastName("lastname");
  expect(instance.state.last_name).toEqual("lastname");
});

it("returns true if zipcode state is changed", () => {
  const instance = setupInstance();
  instance.handleZipCode("55555");
  expect(instance.state.zipcode).toEqual("55555");
});

it("returns error if all fields are not filled", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email@hello.com",
      "password",
      "password",
      "",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if email length is less than 5 characters long", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "e@.c",
      "password",
      "password",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if email doesnt have one @", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email.com",
      "password",
      "password",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if email doesnt have atleast one .", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email@com",
      "password",
      "password",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if password length is less than 6", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email@com",
      "pass",
      "pass",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if password doesn't match confirmPassword", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email@com",
      "pass",
      "password",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "55555",
      "Bio",
    )
  ).toEqual(false);
});

it("returns error if zipcode length is not 5", () => {
  const instance = setupInstance();
  expect(
    instance.validate(
      "email@com",
      "pass",
      "pass",
      "skills",
      "profession",
      "FirstName",
      "LasrName",
      "4444",
      "Bio",
    )
  ).toEqual(false);
});

//render test
it("renders correctly", () => {
  const props = {
    navigation: {
      naviagte: () => {},
    },
  };
  const wrapper = shallow(
    <MentorRegistration {...props}>Snapshot test!</MentorRegistration>
  );

  expect(wrapper).toMatchSnapshot();
});
