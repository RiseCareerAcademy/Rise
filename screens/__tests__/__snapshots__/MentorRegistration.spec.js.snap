// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(Container)
  style={
    Object {
      "margin": 5,
    }
  }
>
  <Styled(Content)>
    <Styled(Text)
      style={
        Object {
          "color": "red",
        }
      }
    />
    <Styled(Form)>
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 85,
            "borderWidth": 3,
            "height": 170,
            "marginBottom": 15,
            "width": 170,
          }
        }
      />
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          Email
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          Password
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          onChangeText={[Function]}
          secureTextEntry={true}
        />
      </Styled(Item)>
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          Confirm Password
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          onChangeText={[Function]}
          secureTextEntry={true}
        />
      </Styled(Item)>
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          Skills (separated by commas)
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        last={true}
        stackedLabel={true}
      >
        <Styled(Label)>
          Profession
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        last={true}
        stackedLabel={true}
      >
        <Styled(Label)>
          Name
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        last={true}
        stackedLabel={true}
      >
        <Styled(Label)>
          Last Name
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        last={true}
        stackedLabel={true}
      >
        <Styled(Label)>
          zipcode
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
      <Styled(Item)
        last={true}
        stackedLabel={true}
      >
        <Styled(Label)>
          Biography
        </Styled(Label)>
        <Styled(Input)
          onChangeText={[Function]}
        />
      </Styled(Item)>
    </Styled(Form)>
  </Styled(Content)>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "height": 40,
        "margin": 15,
        "padding": 10,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
        }
      }
    >
       Next 
    </Styled(Text)>
  </TouchableOpacity>
</Styled(Container)>
`;
